src/rgw/rgw_rest_bucket_logging.cc

```diff
@@ -209,6 +209,122 @@ class RGWPutBucketLoggingOp : public RGWDefaultResponseOp {
+  int verify_permission(optional_yield y) override {
+    auto [has_s3_existing_tag, has_s3_resource_tag] = rgw_check_policy_condition(this, s, false);
+    if (has_s3_resource_tag)
+      rgw_iam_add_buckettags(this, s);
+
+    if (!verify_bucket_permission(this, s, rgw::IAM::s3PostBucketLogging)) {
+      return -EACCES;
+    }
+
+    return 0;
+  }
+
+  const char* name() const override { return "post_bucket_logging"; }
+  RGWOpType get_type() override { return RGW_OP_POST_BUCKET_LOGGING; }
+  uint32_t op_mask() override { return RGW_OP_TYPE_WRITE; }
+
+  void execute(optional_yield y) override {
+    op_ret = verify_bucket_logging_params(this, s);
+    if (op_ret < 0) {
+      return;
+    }
+
+    const auto max_size = s->cct->_conf->rgw_max_put_param_size;
+    bufferlist data;
+    std::tie(op_ret, data) = read_all_input(s, max_size, false);
+    if (op_ret < 0) {
+      ldpp_dout(this, 1) << "ERROR: failed to read XML logging payload, ret = " << op_ret << dendl;
+      return;
+    }
+    if (data.length() == 0) {
+      ldpp_dout(this, 1) << "ERROR: XML logging payload missing" << dendl;
+      op_ret = -EINVAL;
+      return;
+    }
+
+    RGWXMLDecoder::XMLParser parser;
+    if (!parser.init()){
+      ldpp_dout(this, 1) << "ERROR: failed to initialize XML parser" << dendl;
+      op_ret = -EINVAL;
+      return;
+    }
+    if (!parser.parse(data.c_str(), data.length(), 1)) {
+      ldpp_dout(this, 1) << "ERROR: failed to parse XML logging payload" << dendl;
+      op_ret = -ERR_MALFORMED_XML;
+      return;
+    }
+
+    std::string bucket_name;
+    try {
+      RGWXMLDecoder::decode_xml("BucketName", bucket_name, &parser, true);
+    } catch (RGWXMLDecoder::err& err) {
+      ldpp_dout(this, 1) << "ERROR: failed to parse XML logging payload. error: " << err << dendl;
+      op_ret = -ERR_MALFORMED_XML;
+      return;
+    }
+
+    if (bucket_name.empty()) {
+      ldpp_dout(this, 1) << "ERROR: bucket not specified" << dendl;
+      op_ret = -EINVAL;
+      return;
+    }
+    std::unique_ptr<rgw::sal::Bucket> bucket;
+    op_ret = driver->load_bucket(this, rgw_bucket(s->bucket_tenant, s->bucket_name),
+                                 &bucket, y);
+    if (op_ret < 0) {
+      ldpp_dout(this, 1) << "ERROR: failed to get bucket '" << s->bucket_name << "', ret = " << op_ret << dendl;
+      return;
+    }
+    const auto& bucket_attrs = bucket->get_attrs();
+    auto iter = bucket_attrs.find(RGW_ATTR_BUCKET_LOGGING);
+    if (iter == bucket_attrs.end()) {
+      ldpp_dout(this, 1) << "WARNING: no logging configured on bucket" << dendl;
+      return;
+    }
+    rgw::bucketlogging::configuration configuration;
+    try {
+      configuration.enabled = true;
+      decode(configuration, iter->second);
+    } catch (buffer::error& err) {
+      ldpp_dout(this, 1) << "ERROR: failed to decode logging attribute '" << RGW_ATTR_BUCKET_LOGGING
+                         << "'. error: " << err.what() << dendl;
+      return;
```

Comment:
do we expect result here to be sucess?
if not we should probably set:`op_ret = -EINVAL`